lib/new_relic/control/instrumentation.rb

```diff
@@ -62,9 +62,42 @@ def _install_instrumentation
           File.join(instrumentation_path, app.to_s, '*.rb')
         @instrumentation_files.each { |pattern| load_instrumentation_files pattern }
         DependencyDetection.detect!
+        ruby_22_deprecation
+        rails_32_deprecation
         ::NewRelic::Agent.logger.info "Finished instrumentation"
       end
     end
+
+    def rails_32_deprecation
+      if defined?(Rails::VERSION) && Gem::Version.new(Rails::VERSION::STRING) <= Gem::Version.new('3.2')
+        deprecation_msg = 'The Ruby Agent is dropping support for Rails 3.2 ' \
+          'in a future major release. Please upgrade your Rails version to continue receiving support. ' \
+
+        Agent.logger.log_once(
+          :warn,
+          :deprecated_rails_version,
+          deprecation_msg
+        )
+
+        ::NewRelic::Agent.record_metric("Supportability/Deprecated/Rails32", 1)
+      end
+    end
+
+    def ruby_22_deprecation
+      if RUBY_VERSION <= '3.2.0'
```

Comment:
- The `'3.2.0'` string doesn't match the `22` in the method name
- Maybe use an early return here instead of a giant `if`?